py3status/modules/usbguard.py

```diff
@@ -0,0 +1,212 @@
+
+        self.parent.new_data = self.parent._manipulate_devices(
+            self.parent.data)
+        self.parent.py3.update()
+
+    def run(self):
+        while not self.parent.killed.is_set():
+            self.parent._init_dbus()
+            self.parent.dbus.subscribe(
+                object=self.parent.dbus_devices,
+                signal='DevicePresenceChanged',
+                signal_fired=self._on_devices_presence_changed)
+
+            self.parent.dbus.subscribe(
+                object=self.parent.dbus_devices,
+                signal='DevicePolicyChanged',
+                signal_fired=self._on_devices_policy_changed)
+
+            self.loop = GLib.MainLoop()
+            self.loop.run()
+
+
+class Py3status:
+    """
+    """
+    button_allow = 1
+    button_block = 3
+    button_reject = None
+    format = u'[USBGuard: {name}]'
+    format_separator = u' | '
+
+    def _init_dbus(self):
+        self.dbus_interface = 'org.usbguard'
+        self.dbus_devices = '/org/usbguard/Devices'
+        self.dbus = SystemBus()
+        self.error = None
+        try:
+            self.proxy = self.dbus.get(self.dbus_interface)
+        except:
+            self.error = Exception(STRING_USBGUARD_DBUS)
+
+    def _init_placeholders(self):
+        available_placeholders = [
+            'id', 'name', 'via_port', 'hash', 'parent_hash', 'serial',
+            'with_interface'
+        ]
+
+        placeholders = {}
+
+        for placeholder in available_placeholders:
+            if self.py3.format_contains(self.format, placeholder):
+                placeholders[placeholder] = None
+
+        placeholders['usbguard_id'] = None
+        return placeholders
+
+    def post_config_hook(self):
+        self._init_dbus()
+        self.data = {}
+        self.new_data = []
+
+        self.placeholders = self._init_placeholders()
+
+        self.permanant_rule = False
+        self.killed = threading.Event()
+        UsbguardListener(self).start()
+
+    def _usbguard_cmd(self, action, usbguard_id):
+        if action and int(usbguard_id):
+            targets = {'allow': 0, 'block': 1, 'reject': 2}
+            if action == 'block':
+                if self.data[usbguard_id]:
+                    del self.data[usbguard_id]
+                    self.new_data = self._manipulate_devices(self.data)
+                    self.py3.update()
+
+            return self.proxy.applyDevicePolicy(
+                usbguard_id, targets[action], self.permanant_rule
+            )
+
+    def kill(self):
+        self.killed.set()
+
+    def on_click(self, event):
+        self.py3.log(event)
+        button = event['button']
+        usbguard_id = event['index']
+        if button == self.button_allow:
+            self._usbguard_cmd('allow', usbguard_id)
+        elif button == self.button_reject:
+            self._usbguard_cmd('reject', usbguard_id)
+        elif button == self.button_block:
+            self._usbguard_cmd('block', usbguard_id)
```

Comment:
(Optional) I admit I don't like this (eg selecting rules via button numbers instead of just clicking where). This part might be a little too crazy... Make boolean placeholders inside `format_*` placeholders for `allow`, `reject`, `block` to be added in `format_device`. Users then can decide how they customize (eg icons or names)... along with optional toggleable permanent rule. If the device have that rule applied, then it should have `False` to allow hiding `allow`, `block`, `reject` inside `format_*`. This likely  would expose some more configs.

Might be stupid. Might be ugly. Might be overkill. :-)

50 Slytherin points.